### ISSUE TITLE
IE11 forces selected cell to stay in viewport

### ISSUE THREAD

**Issue by PenguinOfThunder** [adazzle/react-data-grid#481]
**I'm submitting a ...**  (check one with "x")

```
[X] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md
```

**Current behavior**
Normally, the user can use the vertical scroll bar (or scroll wheel) to bring the full grid into view. However, in IE11, the grid blocks the viewport content from moving past the point where the selected/focused cell is visible. The only workaround is to click in another field or to change the selection.

This _appears_ to be related to an issue with checkFocus(), but I am unable to confirm at this time.

**Expected/desired behavior**
The user should be able to scroll independent of selected cells.

**Reproduction of the problem**
I can only reproduce this in IE11. It seems to work in Chrome.

**What is the expected behavior?**
The user should be able to scroll independent of selected cells.

**What is the motivation / use case for changing the behavior?**
For tall grids, users need to be able to quickly navigate up and down a grid.

**Comment by willgates** [comment]
Hi @PenguinOfThunder, thanks for bringing this to our attention. We'll look into this.

**Comment by pmurraybne** [comment]
I just had this issue as well.  Apparently it's a known IE11 bug, that is not going to be fixed...?

https://connect.microsoft.com/IE/feedbackdetail/view/900506/ie11-scrolling-issue

**Comment by PenguinOfThunder** [comment]
I have been looking at this some more. I can reliably reproduce the bug, even on the Basic demo page, by opening the page in Internet Explorer 11, then clicking on a cell in the grid.

If I try to scroll up or down past the point where the selected cell would go off-screen, it stops. I have noticed that sometimes it will allow scrolling past, but then jump back into view after a short period of time.

The short delay made me suspect the onScroll function in the Canvas component, but commenting out the code using the rowScrollTimeout did not appear to have any effect.

I have also noted that if I increase the height of the grid to display all rows at once, it prevents the page itself from scrolling the element out of view also.

The problem is temporarily resolved if the user to clicks inside a textbox on the page. This apparently removes focus from the grid (but the cell remains highlighted), and allows it to scroll. However, that is not a viable workaround for our use case, because the grid is the only visible control on that tab.

I tried to run javascript in the console of IE11 to simulate blurring, changing focus and removing the selection from the document. None of this had any effect. Only physically clicking inside another text field worked.

Any suggestions for a workaround?

**Comment by manmamen** [comment]
I just had this issue as well .And i try to change the code checkFocus() ,set 
`if (focusInGrid) {
	        //ReactDOM.findDOMNode(this).focus();
	      }`

It ables it to scroll in IE10 Temporarily .

**Comment by PenguinOfThunder** [comment]
Thank you @manmamen. I tested you change (edit checkFocus in Cell.js). It works, but it basically turns checkFocus into a no-op, so I don't know what else might break. 

The current version (1.0.62) of *react-data-grid* has a completely rewritten version of the code which also exhibits the same problem. If I apply the same "fix" to that code it will work:
```js
if (cellDOMNode) {
     // cellDOMNode.focus();
}
```

So, what breaks if this function is removed or disabled? Can we make it optional?

**Comment by manmamen** [comment]
@PenguinOfThunder I'm not sure what might break if i  turns checkFocus into a no-op.I try the 'enableRowSelect' and 'enableCellSelect' they all works find.

**Comment by manmamen** [comment]
It looks like the bug was fixed? @willgates

**Comment by PenguinOfThunder** [comment]
About 3 hours after the last release, I tested it. The problem as described in this bug has been fixed. The behavior of scrolling the focused cell is still happening is still exhibited, but it is now possible to avoid it by clicking outside of the grid. After the update, it now blurs the grid (as it should) upon clicking outside it.

I question why it needs to forcibly grab focus at all, but like I said I don't know what else would break if it was disabled. This update at least provides as workaround that we can live with.
